packages/labs/esbuild/esbuild.bzl

```diff
@@ -0,0 +1,332 @@
+    name = "bundle",
+    define = [
+        "process.env.NODE_ENV=\\"production\\""
+    ],
+)
+```
+
+See https://esbuild.github.io/api/#define for more details.""",
+            default = [],
+        ),
+        "deps": attr.label_list(
+            aspects = [module_mappings_aspect, node_modules_aspect],
+        ),
+        "entry_point": attr.label(
+            doc = "A single entry point from which the begin the bundle.",
+            allow_single_file = True,
+        ),
+        "entry_points": attr.label_list(
+            doc = """A list of entry points.
+
+Each entry point will create its own bundle.""",
+            allow_files = True,
+        ),
+        "external": attr.string_list(
+            doc = """The list of packages or files considered to be external to the build.
+
+See https://esbuild.github.io/api/#external for more details.""",
+            default = [],
+        ),
+        "format": attr.string(
+            doc = """The bundle's output format.
+One of "iife", "cjs", or "esm".
+
+See https://esbuild.github.io/api/#format for more details.""",
+        ),
+        "loader": attr.string_list(
+            doc = """A list of loaders to use for bundling.
+
+Example:
+
+```python
+esbuild_bundle(
+    name = "bundle",
+    loader = [
+        ".png=dataurl"
+    ],
+)
+```
+
+See https://esbuild.github.io/api/#loader for more details.""",
+            default = [],
+        ),
+        "metafile": attr.bool(
+            doc = """Whether or not to output a metafile.
+
+This will create a metafile with the name `${name}_esbuild_metafile.json` in
+this build target's output directory.
+
+See https://esbuild.github.io/api/#metafile for more details.""",
+        ),
+        "minify": attr.bool(
+            doc = "Whether or not to minify the output.",
+        ),
+        "outdir": attr.bool(
+            doc = """Whether or not to place output files in a separate directory.
+
+This will use the build target's name as the directory name.
+
+See https://esbuild.github.io/api/#outdir for more details.""",
+        ),
+        "platform": attr.string(
+            doc = """The target platform.
+
+See https://esbuild.github.io/api/#platform for more details.""",
+            default = "browser",
+        ),
+        "sourcemap": attr.string(
+            doc = """The method for generating sourcemaps.
+
+See https://esbuild.github.io/api/#sourcemap for more details.
+
+If you wish to use the default sourcemap behaviour, pass in "default".""",
+            default = "default",
+        ),
+        "splitting": attr.bool(
+            doc = """Whether or not to perform code splitting.
+
+See https://esbuild.github.io/api/#splitting for more details.""",
+        ),
+        "srcs": attr.label_list(
+            doc = "Additional files to include in the bundle",
+            allow_files = True,
+            default = [],
+        ),
+        "tsconfig": attr.label(
+            doc = """The tsconfig file to use for the build.
```

Comment:
Does esbuild require a tsconfig? I didn't think it did but I could be wrong. If its not required, I don't think the rule should make it required